### PR TITLE
[ONNX] Implement Softsign op

### PR DESCRIPTION
**Pull request by RattataKing**

### CODE REVIEW

File: test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir [hunk 1]
```diff
@@ -580,6 +580,15 @@ func.func @test_softmax_negative_axis(%arg0: !torch.vtensor<[3,4,5],f32>) -> !to
 
 // -----
 
+// CHECK-LABEL: func.func @test_softsign
+func.func @test_softsign(%arg0: !torch.vtensor<[3,4,5],f32>) -> !torch.vtensor<[3,4,5],f32> attributes {torch.onnx_meta.ir_version = 10 : si64, torch.onnx_meta.opset_version = 22 : si64, torch.onnx_meta.producer_name = "backend-test", torch.onnx_meta.producer_version = ""} {
+  %none = torch.constant.none
+  %0 = torch.operator "onnx.Softsign"(%arg0) : (!torch.vtensor<[3,4,5],f32>) -> !torch.vtensor<[3,4,5],f32>
```
Comment (kuhar) [inline test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir:586]: Should we also CHECK the generated torch ops?

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp [hunk 2]
```diff
@@ -2366,6 +2366,28 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
...
+  patterns.onOp(
+      "Softsign", 22, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value input;
+        if (binder.tensorOperand(input) ||
+            binder.tensorResultType(resultType)) {
+          return failure();
+        }
+
+        Value abs_x = rewriter.create<Torch::AtenAbsOp>(binder.getLoc(),
```
Comment (kuhar) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp:2378]: nit: mlir uses `thisCase` for the local variable naming convention
```suggestion
        Value absX = rewriter.create<Torch::AtenAbsOp>(binder.getLoc(),
```

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp [hunk 3]
```diff
@@ -2366,6 +2366,28 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
...
+                                                        resultType, input);
+
+        Value constOne = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getI64IntegerAttr(1));
+
+        Value abs_x_plus_one = rewriter.create<Torch::AtenAddScalarOp>(
```
Comment (kuhar) [inline lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp:2384]: also here

**Review by kuhar — CHANGES_REQUESTED** [review]

File: test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir [hunk 4]
```diff
@@ -580,6 +580,21 @@ func.func @test_softmax_negative_axis(%arg0: !torch.vtensor<[3,4,5],f32>) -> !to
...
 // -----
 
+// CHECK-LABEL: func.func @test_softsign
+func.func @test_softsign(%arg0: !torch.vtensor<[3,4,5],f32>) -> !torch.vtensor<[3,4,5],f32> attributes {torch.onnx_meta.ir_version = 10 : si64, torch.onnx_meta.opset_version = 22 : si64, torch.onnx_meta.producer_name = "backend-test", torch.onnx_meta.producer_version = ""} {
+  // CHECK: %[[NONE:.*]] = torch.constant.none
+  // CHECK: %[[ABS:.+]] = torch.aten.abs %arg0 : !torch.vtensor<[3,4,5],f32> -> !torch.vtensor<[3,4,5],f32>
+  // CHECK: %[[INT1:.*]] = torch.constant.int 1
+  // CHECK: %[[RES:.+]] = torch.aten.add.Scalar %[[ABS]], %[[INT1]], %[[INT1]] : !torch.vtensor<[3,4,5],f32>, !torch.int, !torch.int -> !torch.vtensor<[3,4,5],f32>
+  // CHECK: %[[SCALE_T:.*]] = torch.aten.div.Tensor %arg0, %[[RES]] : !torch.vtensor<[3,4,5],f32>, !torch.vtensor<[3,4,5],f32> -> !torch.vtensor<[3,4,5],f32>
+  // CHECK: return %[[SCALE_T]] : !torch.vtensor<[3,4,5],f32>
```
Comment (kuhar) [inline test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir:590]: nit: for consistency, use either `.+` or `*.` in this test case

**Review by kuhar — COMMENTED** [review]
Looks fine to me but someone from the torch-mlir side should take a look too

**Review by rsuderman — APPROVED** [review]

File: test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir (hunk 4 above)
Comment (rsuderman) [reply test/Conversion/TorchOnnxToTorch/simple_ops_q_to_z.mlir:590]: Bah, its find. .+ vs .* happens to be fine.

**Pull request merged**